dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
I thought so too, but `MeshShape` has color properties (colors for the vertices) in it. This function gets called when the color of `VisualAddon` is changed so that `MeshShape` can handle it. By default, it does nothing for other shapes. This function is a compromised solution, so I'm open to any better way for this case.